improver/calibration/dataframe_utilities.py

```diff
@@ -443,25 +519,40 @@ def forecast_dataframe_to_cube(
             units=TIME_COORDS["forecast_reference_time"].units,
         )
 
-        for percentile in sorted(df["percentile"].unique()):
-            perc_coord = DimCoord(
-                np.float32(percentile), long_name="percentile", units="%"
-            )
-            perc_df = time_df.loc[time_df["percentile"] == percentile]
+        for var_val in sorted(time_df[representation_type].unique()):
+            var_df = time_df.loc[time_df[representation_type] == var_val]
+            cf_name = var_df["cf_name"].values[0]
+            if representation_type == "percentile":
+                var_coord = DimCoord(
+                    np.float32(var_val), long_name="percentile", units="%"
+                )
+            else:
```

Comment:
If more representations (eg. threshold) are added in the future, this would result in the new representation being treated as realization. Suggest changing to an explicit check for realization - if another representation does get added in the future, this will fail and then get noticed/fixed.
```suggestion
            elif representation_type == "realization":
```